idaes/core/util/model_diagnostics.py

```diff
@@ -500,64 +493,112 @@ def _find_candidate_eqs(candidates_milp, solver, eq_con_list=None, tee=False):
                         name = eq_con_list[i]
                     ds_[name] = candidates_milp.nu[i]()
                     candidate_eqns.append(i)
-
+                    
             return candidate_eqns, ds_
         else:
             return None, None
-
-    def svd_analysis(self, n_smallest_sv=10):
+        
+    
+    def svd_analysis(self, n_smallest_sv=10, dense=False):
         """
         Perform SVD analysis of the constraint Jacobian
-
+        
         Args:
             n_smallest_sv: number of smallest singular values to compute
-
+            
         Returns:
             Nothing
-
+            
         Actions:
             Stores SVD results in object
-
+        
         """
-
+        
         if self.n_eq > 1:
-
+        
             # Determine the number of singular values to compute
             # The "-1" is needed to avoid an error with svds
             n_sv = min(n_smallest_sv, min(self.n_eq, self.n_var) - 1)
             print("Computing the", n_sv, "smallest singular value(s)")
-
+        
             # Perform SVD
             # Recall J is a n_eq x n_var matrix
             # Thus U is a n_eq x n_eq matrix
             # And V is a n_var x n_var
             # (U or V may be smaller in economy mode)
             # Thus we really only care about U
-            u, s, v = svds(self.jac_eq, k=n_sv, which="SM")
-
+            
+            if dense:
+                u, s, vT = svd(self.jac_eq.todense(),full_matrices=False)
+                u = np.flip(u[:,-n_sv:], axis=1)
+                s = np.flip(s[-n_sv:], axis=0)
+                vT = np.flip(vT[-n_sv:,:], axis=0)
+            else:
+                # svds does not guarantee the order in which it generates
+                # singular values, but typically generates them least-to-greatest.
+                # Maybe the warning is for singular values of nearly equal
+                # magnitude or a similar edge case?
+                u, s, vT = svds(self.jac_eq, k = n_sv, which='SM')#, solver='lobpcg')
+            
             # Save results
             self.u = u
             self.s = s
-            self.v = v
-
+            self.v = vT.transpose()
```

Comment:
Is this transpose valid for both dense=True and False?

Reply:
Yes, both `svd` and `svds` return the transpose (or Hermitian transpose) of the right singular vectors.